### PR TITLE
Audio Multiply (FM synthesis node)

### PR DESCRIPTION
**Pull request by AlexMitchellMus**
A new node for <s>FM Synthesis</s> AM Synthesis - accepts any number of inputs, and multiplies them together. 
Node is stateless.

### CODE REVIEW

**Comment by AlexMitchellMus** [comment]
Hey Steve, I hope this is all up to scratch, just fixed an issue with if and else.

I am also guessing from the code style I have looked at from Vuo you would like human readable C, not super optimised code intended for specific compilers? We leave that up to LLVM right?

File: node/vuo.audio/descriptions/vuo.audio.multiply.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Multiply multiple audio channels into a single channel.
+
+- `Samples` - A list of audio sample buffers, each corresponding to one channel.
```
Comment (smokris) [inline node/vuo.audio/descriptions/vuo.audio.multiply.md:3]: Use em-dash ("—") between port name and description

File: node/vuo.audio/vuo.audio.multiply.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * This code may be modified and distributed under the terms of the MIT License.
+ * For more information, see http://vuo.org/license.
+ */
+
+#include "node.h"
+
+VuoModuleMetadata({
+                     "title" : "Multiply Audio Channels",
+                     "keywords" : [ "sound", "music", "merge", "combine", "AM Synthesis", "multiply" ],
+                     "version" : "0.0.1",
```
Comment (smokris) [inline node/vuo.audio/vuo.audio.multiply.c:15]: Node versions should start at 1.0.0.

File: node/vuo.audio/vuo.audio.multiply.c [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                     "node": {
+                        "exampleCompositions" : [ ]
```
Comment (smokris) [inline node/vuo.audio/vuo.audio.multiply.c:17]: Please include an example composition

File: node/vuo.audio/vuo.audio.multiply.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                     }
+                 });
+
+void nodeEvent
+(
+        VuoInputData(VuoList_VuoAudioSamples) samples,
+        VuoOutputData(VuoAudioSamples) multipliedSamples
+)
+{
+  unsigned int channelCount = VuoListGetCount_VuoAudioSamples(samples);
```
Comment (smokris) [inline node/vuo.audio/vuo.audio.multiply.c:27]: Please indent with tabs, not spaces

File: node/vuo.audio/vuo.audio.multiply.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  *multipliedSamples = VuoAudioSamples_alloc(VuoAudioSamples_bufferSize);
+  
+  (*multipliedSamples).samplesPerSecond = VuoAudioSamples_sampleRate;
+
+//  for(unsigned int n = 0; n < (*multipliedSamples).sampleCount; n++) //clear output buffer not needed due to loading output buffer first value
```
Comment (smokris) [inline node/vuo.audio/vuo.audio.multiply.c]: Please don't include obsolete commented code

File: node/vuo.audio/vuo.audio.multiply.c [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+//       (*multipliedSamples).samples[n] = 0.;
+  
+    bool firstAudioPort = true;
+
+    for(unsigned int i = 0; i < channelCount; i++){
```
Comment (smokris) [inline node/vuo.audio/vuo.audio.multiply.c:38]: In general, please follow the [Ellemtel 1992](http://www.literateprogramming.com/ellemtel.pdf) code style (with the exception that we prefer tabs over spaces).  Braces on their own line, space between keywords and `(`.

File: node/vuo.audio/vuo.audio.multiply.c (hunk 5 above)
Comment (smokris) [inline node/vuo.audio/vuo.audio.multiply.c:33]: If the input list has 0 items, we should still initialize the buffer to 0.

File: node/vuo.audio/vuo.audio.multiply.c [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        VuoAudioSamples as = VuoListGetValue_VuoAudioSamples(samples, i+1);
+
+        if(firstAudioPort && VuoAudioSamples_isPopulated(as)){
```
Comment (smokris) [inline node/vuo.audio/vuo.audio.multiply.c:41]: No need to check `_isPopulated`, since the conditional in the following `for` blocks does the same thing.

**Comment by smokris** [comment]
Thanks, @alcomposer.  I provided some feedback on specific lines above.  Could you fix those issues, then submit this as a feature request, per the new procedure on https://vuo.org/contribute/code ?

**Comment by AlexMitchellMus** [comment]
ok

**Pull request closed without merging**